Bottom: 585a7798e044986ccfb1fe56fe36eea245ea4895
Top:    3c9175919afa0c3a7589f948481d92519a7ba4ab
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-11 15:33:25 +0000

Fixup gtkhid-gl to build


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 27cd75d..112ab4d 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -281,13 +281,13 @@ ghid_set_layer (const char *name, int group, int empty)
 	}
     }
 
-  end_subcomposite ();
+  end_subcomposite (hidgl);
 
   if (group_visible && subcomposite)
-    start_subcomposite ();
+    start_subcomposite (hidgl);
 
   /* Drawing is already flushed by {start,end}_subcomposite */
-  hidgl_set_depth (compute_depth (group));
+  set_depth_on_all_active_gc (priv, compute_depth (group));
 
   return group_visible;
 }
@@ -1406,6 +1406,7 @@ GhidDrawMask (int side, BoxType * screen)
 static int
 GhidDrawLayerGroup (int group, const BoxType * screen)
 {
+  render_priv *priv = gport->render_priv;
   int i, rv = 1;
   int layernum;
   int side;
@@ -1438,13 +1439,13 @@ GhidDrawLayerGroup (int group, const BoxType * screen)
 
       if (rv && !TEST_FLAG (THINDRAWFLAG, PCB)) {
         /* Mask out drilled holes on this layer */
-        hidgl_flush_triangles (&buffer);
+        hidgl_flush_triangles (priv->hidgl);
         glPushAttrib (GL_COLOR_BUFFER_BIT);
         glColorMask (0, 0, 0, 0);
         hid_draw_set_color (Output.bgGC, PCB->MaskColor);
         if (PCB->PinOn) r_search (PCB->Data->pin_tree, screen, NULL, hole_callback, NULL);
         if (PCB->ViaOn) r_search (PCB->Data->via_tree, screen, NULL, hole_callback, NULL);
-        hidgl_flush_triangles (&buffer);
+        hidgl_flush_triangles (priv->hidgl);
         glPopAttrib ();
       }
 
@@ -1460,13 +1461,13 @@ GhidDrawLayerGroup (int group, const BoxType * screen)
         gui->set_layer (0, group, 0);
 
         if (rv && !TEST_FLAG (THINDRAWFLAG, PCB)) {
-          hidgl_flush_triangles (&buffer);
+          hidgl_flush_triangles (priv->hidgl);
           glPushAttrib (GL_COLOR_BUFFER_BIT);
           glColorMask (0, 0, 0, 0);
           /* Mask out drilled holes on this layer */
           if (PCB->PinOn) r_search (PCB->Data->pin_tree, screen, NULL, hole_callback, NULL);
           if (PCB->ViaOn) r_search (PCB->Data->via_tree, screen, NULL, hole_callback, NULL);
-          hidgl_flush_triangles (&buffer);
+          hidgl_flush_triangles (priv->hidgl);
           glPopAttrib ();
         }
       }
@@ -1502,7 +1503,7 @@ GhidDrawLayerGroup (int group, const BoxType * screen)
 }
 
 static void
-DrawDrillChannel (int vx, int vy, int vr, int from_layer, int to_layer, double scale)
+DrawDrillChannel (hidGC gc, int vx, int vy, int vr, int from_layer, int to_layer, double scale)
 {
 #define PIXELS_PER_CIRCLINE 5.
 #define MIN_FACES_PER_CYL 6
@@ -1528,13 +1529,13 @@ DrawDrillChannel (int vx, int vy, int vr, int from_layer, int to_layer, double s
   x1 = vx + vr;
   y1 = vy;
 
-  hidgl_ensure_triangle_space (&buffer, 2 * slices);
+  hidgl_ensure_triangle_space (gc, 2 * slices);
   for (i = 0; i < slices; i++)
     {
       x2 = radius * cosf (((float)(i + 1)) * 2. * M_PI / (float)slices) + vx;
       y2 = radius * sinf (((float)(i + 1)) * 2. * M_PI / (float)slices) + vy;
-      hidgl_add_triangle_3D (&buffer, x1, y1, z1,  x2, y2, z1,  x1, y1, z2);
-      hidgl_add_triangle_3D (&buffer, x2, y2, z1,  x1, y1, z2,  x2, y2, z2);
+      hidgl_add_triangle_3D (gc, x1, y1, z1,  x2, y2, z1,  x1, y1, z2);
+      hidgl_add_triangle_3D (gc, x2, y2, z1,  x1, y1, z2,  x2, y2, z2);
       x1 = x2;
       y1 = y2;
     }
@@ -1563,7 +1564,7 @@ draw_hole_cyl (PinType *Pin, struct cyl_info *info, int Type)
     color = "drill";
 
   hid_draw_set_color (Output.fgGC, color);
-  DrawDrillChannel (Pin->X, Pin->Y, Pin->DrillingHole / 2, info->from_layer, info->to_layer, info->scale);
+  DrawDrillChannel (Output.fgGC, Pin->X, Pin->Y, Pin->DrillingHole / 2, info->from_layer, info->to_layer, info->scale);
   return 0;
 }
 
@@ -1694,16 +1695,16 @@ ghid_draw_everything (BoxType *drawn_area)
 
   /* Draw pins, pads, vias below silk */
   if (global_view_2d) {
-    start_subcomposite ();
+    start_subcomposite (priv->hidgl);
 
     if (!TEST_FLAG (THINDRAWFLAG, PCB)) {
       /* Mask out drilled holes */
-      hidgl_flush_triangles (&buffer);
+      hidgl_flush_triangles (priv->hidgl);
       glPushAttrib (GL_COLOR_BUFFER_BIT);
       glColorMask (0, 0, 0, 0);
       if (PCB->PinOn) r_search (PCB->Data->pin_tree, drawn_area, NULL, hole_callback, NULL);
       if (PCB->ViaOn) r_search (PCB->Data->via_tree, drawn_area, NULL, hole_callback, NULL);
-      hidgl_flush_triangles (&buffer);
+      hidgl_flush_triangles (priv->hidgl);
       glPopAttrib ();
     }
 
@@ -1711,7 +1712,7 @@ ghid_draw_everything (BoxType *drawn_area)
     if (PCB->PinOn) r_search (PCB->Data->pin_tree, drawn_area, NULL, pin_callback, NULL);
     if (PCB->ViaOn) r_search (PCB->Data->via_tree, drawn_area, NULL, via_callback, NULL);
 
-    end_subcomposite ();
+    end_subcomposite (priv->hidgl);
   }
 
   /* Draw the solder mask if turned on */
@@ -1939,7 +1940,7 @@ ghid_drawing_area_expose_cb (GtkWidget *widget,
   draw_crosshair (Output.fgGC, priv);
   hidgl_flush_triangles (priv->hidgl);
 
-  draw_lead_user (priv->crosshair_gc, priv);
+  draw_lead_user (Output.fgGC, priv);
 
   ghid_end_drawing (port, widget);
